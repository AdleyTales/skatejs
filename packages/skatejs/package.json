{
  "name": "skatejs",
  "description": "Skate is a library built on top of the W3C web component specs that enables you to write functional and performant web components with a very small footprint.",
  "author": "Trey Shugart <treshugart@gmail.com> (http://treshugart.github.io)",
  "license": "MIT",
  "keywords": [
    "components",
    "custom",
    "dom",
    "elements",
    "web",
    "vdom",
    "virtual"
  ],
  "repository": "https://github.com/skatejs/skatejs",
  "version": "5.1.1",
  "files": [
    "dist/cjs",
    "dist/esm"
  ],
  "browser": "dist/esm/index.js",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "devDependencies": {
    "@skatejs/bore": "4.0.3"
  }
}
