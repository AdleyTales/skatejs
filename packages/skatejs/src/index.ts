export * from './component';
export * from './define';
export * from './name';
export * from './props';
export * from './types';
