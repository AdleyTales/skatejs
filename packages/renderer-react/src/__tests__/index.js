import React, { Component } from 'react';
import { define } from 'skatejs';
import withRenderer, { asCustomElement } from '..';

function render(Comp) {
  const el = new Comp();
  el.renderer(el, el.render.bind(el));
  return el;
}

class ReactComponent extends Component {
  render() {
    return <div>Hello, {this.props.children}!</div>;
  }
}

test('renders', () => {
  @define
  class MyElement extends withRenderer() {
    render({ name }) {
      return <div>Hello, {name}!</div>;
    }
  }

  const el = new MyElement();
  expect(el.innerHTML).toMatchSnapshot();
  el.renderer(el, el.render.bind(el, { name: 'World' }));
  expect(el.innerHTML).toMatchSnapshot();
  el.renderer(el, el.render.bind(el, { name: 'Bob' }));
  expect(el.innerHTML).toMatchSnapshot();
});

test('wrapper - static component', () => {
  const ReactComponentWrapper = asCustomElement(ReactComponent);

  define(ReactComponentWrapper);
  const el = render(ReactComponentWrapper);
  expect(el.innerHTML).toMatchSnapshot();
});

test('wrapper - override render()', () => {
  @define
  class ReactComponentWrapper extends asCustomElement(ReactComponent) {
    render() {
      return <ReactComponent {...this.props} />;
    }
  }

  const el = new ReactComponentWrapper();
  el.renderer(el, el.render.bind(el));
  expect(el.innerHTML).toMatchSnapshot();
});
