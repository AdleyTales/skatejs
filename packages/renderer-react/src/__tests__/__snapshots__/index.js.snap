// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `""`;

exports[`renders 2`] = `"<div data-reactroot=\\"\\"><!-- react-text: 2 -->Hello, <!-- /react-text --><!-- react-text: 3 -->World<!-- /react-text --><!-- react-text: 4 -->!<!-- /react-text --></div>"`;

exports[`renders 3`] = `"<div data-reactroot=\\"\\"><!-- react-text: 2 -->Hello, <!-- /react-text --><!-- react-text: 3 -->Bob<!-- /react-text --><!-- react-text: 4 -->!<!-- /react-text --></div>"`;

exports[`unmounts 1`] = `"<div data-reactroot=\\"\\"></div>"`;

exports[`wrapper - override render() 1`] = `"<div data-reactroot=\\"\\"><!-- react-text: 2 -->Hello, <!-- /react-text --><slot></slot><!-- react-text: 4 -->!<!-- /react-text --></div>"`;

exports[`wrapper - static component 1`] = `"<div data-reactroot=\\"\\"><!-- react-text: 2 -->Hello, <!-- /react-text --><slot></slot><!-- react-text: 4 -->!<!-- /react-text --></div>"`;
